ipq806x: replace patch with upstream version

Ansuel opened the pull request:
- Replace all the custom patch with the backported upstream version.
- Rework the dts to use the upstream bindings.
- Update changed config flags

Bindings changes are:
- PCI new compatible to handle tx term offset difference
- force_gen to standard max-speed param
- change of name for usb phy dwc3 driver
- change of bindings for usb phy params (mpll fixed to the right value)
- cpufreq nvmem driver to the upstream name

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
@adrianschmutzler @mans0n Sorry if i quote you but can someone review this and check if i don't messed up something? 
I'm stupid and I had to do this earlier to not push a massive PR...

adrianschmutzler commented:
It would certainly help if you split that patch down into logical units, i.e. based on the bindings/properties you change, so that each commit only touches one property and one or a few patches, if possible.

Ansuel commented:
Sure

Ansuel commented:
@adrianschmutzler should I keep the DWC3 driver name or i should change it to the config name? Also i notice that it's displayed for ipq40x but i can't find any dts that use it

Ansuel committed:
ipq806x: replace pci patchset with upstream version

Changes:
- Update patch
- Update dts with new binding
    Tx term offset dropped and replaced with a new compatible

Removed:
- 0071-5-PCI-qcom-Programming-the-PCIE-iATU-for-IPQ806x
    Pci init does the same exact thing (was needed in older kernel version)
- 0071-7-pcie-Set-PCIE-MRRS-and-MPS-to-256B
    Rejected upstream, can't find any reason to have this. No regression with
    testing it on R7800.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

Ansuel commented:
This should be better

adrianschmutzler commented:
I picked the first one and will merge it later today, but I lack the deeper understanding to properly review the other ones.

Ansuel commented:
Thx hope the others gets also reviewed and this doesn't get lost

adrianschmutzler commented:
I can be bribed with test results and/or reviews from non-committers as well.

adrianschmutzler commented:
@Ansuel Consider rebasing onto master so only the three relevant commits remain.

adrianschmutzler commented on target/linux/ipq806x/patches-5.4/083-ipq8064-dtsi-additions.patch:
```diff
@@ -67,7 +67,7 @@
 +
 +	opp_table0: opp_table0 {
-+		compatible = "operating-points-v2-qcom-cpu";
++		compatible = "operating-points-v2-kyro-cpu";
```
I read "kryo" in the patch, not kyro ...
Can you provide some details on why that naming was changed in the first place, and now is going back to kryo again ...?

adrianschmutzler replied:
And why does the upstream commit:
```
-	ret = of_device_is_compatible(np, "operating-points-v2-kryo-cpu");
+	ret = of_device_is_compatible(np, "operating-points-v2-qcom-cpu");
```

which is removed here?

Ansuel replied:
The first one is a typo... 
The name reverted is for compatible reason
(We can't change compatible name of already pushed driver)

adrianschmutzler replied:
Then please state it in the patch. Currently, you create the impression that this is an upstream patch, while it actually is an altered upstream patch. I shouldn't need to compare manually with upstream to find that out; do you want me to check every patch line-by-line for whether it's _really_ upstream?

I'd rather prefer to take the upstream patch as is and then add locally required adjustments in a separate patch, so it immediately obvious what's upstream and what's not (where that is possible considering backporting as well).

Ansuel replied:
Patch upstream is this
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/cpufreq/qcom-cpufreq-nvmem.c?id=a8811ec764f95a04ba82f6f457e28c5e9e36e36b
and right after there is
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/cpufreq/qcom-cpufreq-nvmem.c?id=2dea651680cea1f3a29925de51002f33d1f55711

I combined them together since it looks like a waste to have the second patch that do this change. (only a revert of the compatible)

The only other patch that is changed from the upstream version is 
- pci patchset that lacks of one include the in 8th patch 
- usb dwc3 patch that has a small fix (same thing a waste to have a second patch that fix one line)

If you want i can put restore this and add the 2 missing patch that are combined

Ansuel replied:
@adrianschmutzler  I added the deviation from the upstream patch in the patch description. Is it good now?

adrianschmutzler replied:
As you have seen, this is an unnecessary reason for confusion. Why merge these particular patches and still keep the illusion of them being "real" upstream patches? If we locally mess with the patches again, then the whole "replace local patches by upstream thing" is just useless waste of time. For me, being an upstream patch implies that I don't have to think about it again.
So, please include the patches in their upstream version unless backporting needs adjustments. For local adjustments, create separate patches.

Ansuel replied:
Ok sorry for the mess
I will follow this path

mans0n commented:
It seems the dwc3 driver was just copied & pasted from ipq806x when ipq40xx target was newly introduced. https://github.com/openwrt/openwrt/commit/54b275c8ed3ad20c447fd46deec83384822ac79d#diff-008ef19a0aea6e5fb9be559aae12c2c3R19
Note that usb-phy-qcom-dwc3 in the commit above still depends on `@TARGET_ipq806x`, which does not sense for ipq40xx target.
And https://github.com/openwrt/openwrt/commit/1b906723ebea2319ba428ae708867273805ddd43#diff-4f8ed299ce19aab7575b21850d9b41fdR76 moved the modules to generic, which also added `@TARGET_ipq40xx` to the driver.

I guess we can just remove `@TARGET_ipq40xx` from DEPENDS, and perhaps move it to ipq806x modules.mk.

Ansuel commented:
@mans0n added your suggestion to the last commit

adrianschmutzler commented on target/linux/ipq806x/patches-5.4/095-ipq806x-phy-qualcomm-add-qcom-ipq806x-dwc-usb-phy-driver.patch:
```diff
@@ -0,0 +1,632 @@
+Signed-off-by: Vinod Koul <vkoul@kernel.org>
+---
+
+Combined with 3d7b0ca5300bd01b176f2b4c10e173db802560d8
```
I'd like separate patches here as well.

adrianschmutzler commented on package/kernel/linux/modules/usb.mk:
```diff
@@ -73,10 +73,10 @@ $(eval $(call KernelPackage,usb-phy-nop))
-    $(LINUX_DIR)/drivers/phy/qualcomm/phy-qcom-dwc3.ko
-  AUTOLOAD:=$(call AutoLoad,45,phy-qcom-dwc3,1)
+    $(LINUX_DIR)/drivers/phy/qualcomm/phy-qcom-ipq806x_usb.ko
+  AUTOLOAD:=$(call AutoLoad,45,phy-qcom-ipq806x_usb,1)
```
Is the underscore correct here? Or should it be phy-qcom-ipq806x-usb ?

adrianschmutzler commented on package/kernel/linux/modules/usb.mk:
```diff
@@ -73,10 +73,10 @@ $(eval $(call KernelPackage,usb-phy-nop))
 define KernelPackage/usb-phy-qcom-dwc3
```
I'd personally already rename the kmod here as well, in sync with the kconfig/ko file, and only move it to ipq806x target in that separate patch.

adrianschmutzler commented:
With the requested adjustments, I'd consider merging everything except the pci patch.

Ansuel commented:
LGTM I also want extensive test for the pci patch... Will ask on the forum

Ansuel commented on target/linux/ipq806x/patches-5.4/095-ipq806x-phy-qualcomm-add-qcom-ipq806x-dwc-usb-phy-driver.patch:
```diff
@@ -0,0 +1,632 @@
+---
+
+Combined with 3d7b0ca5300bd01b176f2b4c10e173db802560d8
+Light modification to Kconfig as some config are missing in kernel 5.4
```
The patch doesn't apply without this modification to kconfig and makefile... Can i leave this?

adrianschmutzler replied:
The modifications are fine. Of course, backporting is necessary, and we cannot just add a patch from a newer kernel as-is in any case :-)

Ansuel commented:
@adrianschmutzler except for that comment all should be good

adrianschmutzler commented:
Hi, I've found another typo with qcom,tx-deamp_3_5db

Anyway, I've updated and pushed the relevant patches here:
https://git.openwrt.org/?p=openwrt/staging/adrian.git;a=shortlog;h=refs/heads/ipq806x

Since we already had a few typos on the way, I'd be happy if we could one or a few tested-by for those before merging.

Ansuel commented:
@adschm think for the pci we need more testing just to make sure since it's a bit critical. I have at least 4 guy that tested the patch but I would like to test this on other router. Any idea how to contact other testers?

adschm commented:
> @adschm think for the pci we need more testing just to make sure since it's a bit critical. I have at least 4 guy that tested the patch but I would like to test this on other router. Any idea how to contact other testers?

Look in the GitHub history who added/changed other routers (dts, config) and highlight them here with a short explanation.

adschm commented:
These "(limited to 'drivers/pci/controller/dwc/pcie-qcom.c')" lines in the patch headers appear to be a remnant of the patch creation? At least, the one where I checked only touches the single file in the original anyway ...

Ansuel commented:
Will fix them asap (the patch only touch pcie-qcom file but still looks bad)

Il giorno gio 13 ago 2020 alle ore 12:10 Adrian Schmutzler <
notifications@github.com> ha scritto:

> These "(limited to 'drivers/pci/controller/dwc/pcie-qcom.c')" lines in the
> patch headers appear to be a remnant of the patch creation? At least, the
> one where I checked only touches the single file in the original anyway ...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/3284#issuecomment-673390248>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQUANSHHOCQFHJYREJDSAO32LANCNFSM4PYVNDEA>
> .
>

adschm commented:
Thanks to all participants, I think this is good enough to merge it.

CHKDSK88 commented:
@adschm 

In Asrock G10 current master work ok.
